lib/CompaniesHouseGenericService.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright (c) 2017, Will Evans.
+ * Licensed under the MIT License.
+ */
+const axios = require('axios');
+
+class CompaniesHouseGenericService {
+
+    constructor(apiKey){
+        this.apiKey = apiKey;
+    }
+
+    /**
+     * This method calls the companies house API to return data for a specific link e.g one returned in Company Profile
+     * @param link
+     * @return {Promise}
+     */
+    getLink(link){
+        return new Promise(
+            (resolve, reject) => {
+                axios({
+                    method:'get',
+                    url: `https://api.companieshouse.gov.uk${link}`,
```

Comment:
Link is a link returned by the gov in a document. Always will append correctly. Say if you get a company profile you'll get links to officers, filing history etc.